packages/button/index.js

```diff
@@ -63,64 +63,70 @@ class WarpButton extends kebabCaseAttributes(WarpElement) {
     }
   }
 
+  get _primaryClasses() {
+    return [
+      !this.small && !this.quiet && !this.loading && ccButton.primary,
+      this.small && !this.quiet && !this.loading && ccButton.primarySmall,
+      this.small && this.quiet && !this.loading && ccButton.primarySmallQuiet,
+      this.small && this.loading && (this.quiet ? ccButton.primarySmallQuietLoading : ccButton.primarySmallLoading),
+      !this.small && this.quiet && !this.loading && ccButton.primaryQuiet,
+      !this.small && this.loading && (this.quiet ? ccButton.primaryQuietLoading : ccButton.primaryLoading),
+    ];
+  }
+
+  get _secondaryClasses() {
+    return [
+      !this.small && !this.quiet && !this.loading && ccButton.secondary,
+      this.small && !this.quiet && !this.loading && ccButton.secondarySmall,
+      this.small && this.loading && (this.quiet ? ccButton.secondarySmallQuietLoading : ccButton.secondarySmallLoading),
+      this.small && this.quiet && !this.loading && ccButton.secondarySmallQuiet,
+      !this.small && this.quiet && !this.loading && ccButton.secondaryQuiet,
+      !this.small && this.loading && (this.quiet ? ccButton.secondaryQuietLoading : ccButton.secondaryLoading),
+    ];
+  }
+
+  get _utilityClasses() {
+    return [
+      !this.small && !this.quiet && !this.loading && ccButton.utility,
+      this.small && !this.quiet && !this.loading && ccButton.utilitySmall,
+      this.small && this.quiet && !this.loading && ccButton.utilitySmallQuiet,
+      this.small && this.loading && (this.quiet ? ccButton.utilitySmallQuietLoading : ccButton.utilitySmallLoading),
+      !this.small && this.quiet && !this.loading && ccButton.utilityQuiet,
+      !this.small && this.loading && (this.quiet ? ccButton.utilityQuietLoading : ccButton.utilityLoading),
+    ];
+  }
+
+  get _negativeClasses() {
+    return [
+      !this.small && !this.quiet && !this.loading && ccButton.negative,
+      this.small && !this.quiet && !this.loading && ccButton.negativeSmall,
+      this.small && this.quiet && !this.loading && ccButton.negativeSmallQuiet,
+      this.small && this.loading && (this.quiet ? ccButton.negativeSmallQuietLoading : ccButton.negativeSmallLoading),
+      !this.small && this.quiet && !this.loading && ccButton.negativeQuiet,
+      !this.small && this.loading && (this.quiet ? ccButton.negativeQuietLoading : ccButton.negativeLoading),
+    ];
+  }
+
+  get _pillClasses() {
+    return [
+      !this.loading && (this.small ? ccButton.pillSmall : ccButton.pill),
+      this.loading && (this.small ? ccButton.pillSmallLoading : ccButton.pillLoading),
+    ];
+  }
+
+  get _linkClasses() {
+    return [this.small ? ccButton.linkSmall : ccButton.link, !!this.href && ccButton.linkAsButton];
```

Comment:
I found a problem with this solution because we never enter this line when `link` is `false` while `href` has value:
<img width="147" alt="Screenshot of a broken link that looks like a secondary button" src="https://github.com/user-attachments/assets/81c9f4a0-c8eb-43c4-b9a9-ae9316c4fdbc">

We need to move `!!this.href && ccButton.linkAsButton` to line 128, and we can have `this.href && ccButton.linkAsButton` there.
<img width="167" alt="Screenshot of a correct looking link that looks like a secondary button" src="https://github.com/user-attachments/assets/ec262463-98ee-48d8-87f6-331038bef911">

Reply:
Ah nicely spotted! 🙏  I'll move that part! 🎉

Reply:
Updated in this commit: [04a29c8a](https://github.com/warp-ds/elements/pull/174/commits/04a29c8aadaf9281cd1cdc8a627548ed6cc3269a)